e2e/test/MessageReceiveE2ETests.cs

```diff
@@ -28,63 +32,147 @@ public MessageReceiveE2ETests()
+            await ReceiveSingleMessage(TestDeviceType.Sasl, Client.TransportType.Amqp_Tcp_Only, VerifyReceivedC2DMessageAndReject).ConfigureAwait(false);
         }
 
         [TestMethod]
-        public async Task Message_DeviceReceiveSingleMessage_Mqtt()
+        public async Task Message_DeviceReceiveSingleMessageComplete_AmqpWs()
         {
-            await ReceiveSingleMessage(TestDeviceType.Sasl, Client.TransportType.Mqtt_Tcp_Only).ConfigureAwait(false);
+            await ReceiveSingleMessage(TestDeviceType.Sasl, Client.TransportType.Amqp_WebSocket_Only, VerifyReceivedC2DMessageAndComplete).ConfigureAwait(false);
+        }
+
+        [TestMethod]
+        public async Task Message_DeviceReceiveSingleMessageAbandon_AmqpWs()
+        {
+            await ReceiveSingleMessage(TestDeviceType.Sasl, Client.TransportType.Amqp_WebSocket_Only, VerifyReceivedC2DMessageAndAbandon).ConfigureAwait(false);
+        }
+
+        [TestMethod]
+        public async Task Message_DeviceReceiveSingleMessageReject_AmqpWs()
+        {
+            await ReceiveSingleMessage(TestDeviceType.Sasl, Client.TransportType.Amqp_WebSocket_Only, VerifyReceivedC2DMessageAndReject).ConfigureAwait(false);
+        }
+
+        [TestMethod]
+        public async Task Message_DeviceReceiveSingleMessageComplete_Mqtt()
+        {
+            await ReceiveSingleMessage(TestDeviceType.Sasl, Client.TransportType.Mqtt_Tcp_Only, VerifyReceivedC2DMessageAndComplete).ConfigureAwait(false);
         }
 
         [TestMethod]
         public async Task Message_DeviceReceiveSingleMessage_MqttWs()
         {
-            await ReceiveSingleMessage(TestDeviceType.Sasl, Client.TransportType.Mqtt_WebSocket_Only).ConfigureAwait(false);
+            await ReceiveSingleMessage(TestDeviceType.Sasl, Client.TransportType.Mqtt_WebSocket_Only, VerifyReceivedC2DMessageAndComplete).ConfigureAwait(false);
+        }
+
+        [TestMethod]
+        public async Task Message_DeviceReceiveSingleMessageComplete_Http()
+        {
+            await ReceiveSingleMessage(TestDeviceType.Sasl, Client.TransportType.Http1, VerifyReceivedC2DMessageAndComplete).ConfigureAwait(false);
+        }
+
+        [TestMethod]
+        public async Task Message_DeviceReceiveSingleMessageAbandon_Http()
+        {
+            await ReceiveSingleMessage(TestDeviceType.Sasl, Client.TransportType.Http1, VerifyReceivedC2DMessageAndAbandon).ConfigureAwait(false);
         }
 
         [TestMethod]
-        public async Task Message_DeviceReceiveSingleMessage_Http()
+        public async Task Message_DeviceReceiveSingleMessageReject_Http()
         {
-            await ReceiveSingleMessage(TestDeviceType.Sasl, Client.TransportType.Http1).ConfigureAwait(false);
+            await ReceiveSingleMessage(TestDeviceType.Sasl, Client.TransportType.Http1, VerifyReceivedC2DMessageAndReject).ConfigureAwait(false);
         }
 
         [TestMethod]
         public async Task X509_DeviceReceiveSingleMessage_Amqp()
```

Comment:
nit: Naming convention - these tests should also be renamed to indicate if the receive operation completes/is rejected/ is abandoned.

Reply:
Test name already reflects Reject/abandon etc. ReceiveSingleMessage is still relevant here as the test is receiving one message and then reject/abandon/complete it. Let me know if I misunderstood your comment.